### ISSUE TITLE
Explorer plugin - Everything: custom filters

### ISSUE THREAD

**Issue by gisturbance** [Flow-Launcher/Flow.Launcher#2009]
**Is your feature request related to a problem? Please describe.**

Sometimes when searching, it can be useful to limit search within certain criteria:
- excluding all files that are offline (on a disconnected drive)
- searching within audio-only files (since different file extensions extist)
- searching within files smaller than 10mb

In Everythings GUI, this is possible with **filters**, that utilize **functions**. Currently in 1.5a, **filters are not being exposed** through the SDK, but **functions are**.

Functions look like that: `ext:mp3|txt|jpg size:>10kb size:<5mb online:`
Give it a name and you've got a filter.

**Describe the solution you'd like**

A solution would be to manually implement filters.
Consisting of a name and a string of functions, a filter just has to be prepended to the search string (with an empty space between).

Example:
Name: `audio` - Content: `ext:mp3|flac|opus size:<3mb online:`
Usage: `audio: <some_search_string>`

Additionally, one filter should be markable as default, always being applied when omitting its name (unless another filter is used).

**Describe alternatives you've considered**

The alternative is to manually specify functions with every search.

**Comment by deefrawley** [comment]
This could be done if there was an option in Custom Query Shortcuts to allow the shortcut to append or prepend some text rather than be the entire query.
@gisturbance - a workaround in the meantime could be to use something like Autohotkey or Espanso to create a text shortcut for the filter, type that in the Flow window and let those apps auto expand the text.

**Comment by gisturbance** [comment]
@deefrawley Using the methods you suggested, a default filter would still be missing, though one could use a different keyboard combination as a workaround for that.

For whatever reason, the Custom Query Shortcuts aren't working for me - they don't expand, while the built-in ones do.

**Comment by DRYN07** [comment]
Any news about this? This feature would be very useful. Additionally we could specify search tags and define what to search for.

![image](https://github.com/Flow-Launcher/Flow.Launcher/assets/29770761/c8f6fe74-0b78-4186-985a-313785d41ca7)

**Comment by jjw24** [comment]
Unfortunately not aware anyone that has started working on this.

**Comment by deefrawley** [comment]
Since my first answer I've found you can use custom query shortcuts to set these up. 

![image](https://github.com/Flow-Launcher/Flow.Launcher/assets/18594055/699087c9-9e45-456a-a4cd-bca12a1f7fe6)

Use the at symbol after the search query to invoke

![image](https://github.com/Flow-Launcher/Flow.Launcher/assets/18594055/1bb4fcc5-bc54-4701-815d-3ecf6e24ca4c)

**Comment by DRYN07** [comment]
Yep, I just found the exact solution you mentioned a few days before. It works perfectly, thanks for sharing it here as well.

**Comment by gisturbance** [comment]
> Use the at symbol after the search query to invoke

That's what I've always been using, though writing the `@` isn't quite appealing, even if you can combine it with Custom Query Hotkeys...
